vhdbuilder/packer/buildperformance/pkg/service/helper.go

```diff
@@ -0,0 +1,198 @@
+package service
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+)
+
+// Set conditions for program to run successfully wtih SetupConfig and CreateDataMaps functions
+func SetupConfig() (*Config, error) {
+	envVars := map[string]string{
+		"kustoTable":                os.Getenv("BUILD_PERFORMANCE_TABLE_NAME"),
+		"kustoEndpoint":             os.Getenv("BUILD_PERFORMANCE_KUSTO_ENDPOINT"),
+		"kustoDatabase":             os.Getenv("BUILD_PERFORMANCE_DATABASE_NAME"),
+		"kustoClientId":             os.Getenv("BUILD_PERFORMANCE_CLIENT_ID"),
+		"kustoIngestionMapping":     os.Getenv("BUILD_PERFORMANCE_INGESTION_MAPPING"),
+		"sourceBranch":              os.Getenv("GIT_BRANCH"),
+		"sigImageName":              os.Getenv("SIG_IMAGE_NAME"),
+		"localBuildPerformanceFile": os.Getenv("SIG_IMAGE_NAME") + "-build-performance.json",
+	}
+	missingVar := false
+	for name, value := range envVars {
+		if value == "" {
+			fmt.Printf("missing environment variable \"%s\".", name)
+			missingVar = true
+		}
+	}
+	if missingVar {
+		return nil, fmt.Errorf("required environment variables were not set")
+	}
+
+	return &Config{
+		KustoTable:                envVars["kustoTable"],
+		KustoEndpoint:             envVars["kustoEndpoint"],
+		KustoDatabase:             envVars["kustoDatabase"],
+		KustoClientId:             envVars["kustoClientId"],
+		KustoIngestionMapping:     envVars["kustoIngestionMapping"],
+		SigImageName:              envVars["sigImageName"],
+		LocalBuildPerformanceFile: envVars["localBuildPerformanceFile"],
+		SourceBranch:              envVars["sourceBranch"],
+	}, nil
+}
+
+func CreateDataMaps() *DataMaps {
+	return &DataMaps{
+		// LocalPerformanceDataMap will hold the performance data from the local JSON file
+		LocalPerformanceDataMap: map[string]map[string]float64{},
+		// QueriedPerformanceDataMap will hold the aggregated performance data queried from Kusto
+		QueriedPerformanceDataMap: map[string]map[string][]float64{},
+		// RegressionMap will hold all identified regressions in the current build
+		RegressionMap: map[string]map[string]float64{},
+	}
+}
+
+// Prepare performance data for evaluation with PreparePerformanceDataForEvaluation and associated helper functions
+func (maps *DataMaps) PreparePerformanceDataForEvaluation(localBuildPerformanceFile string, queriedData *SKU) error {
+	err := maps.DecodeLocalPerformanceData(localBuildPerformanceFile)
+	if err != nil {
+		return fmt.Errorf("error decoding local performance data: %w", err)
+	}
+	err = maps.ParseKustoData(queriedData)
+	if err != nil {
+		return fmt.Errorf("error parsing kusto data: %w", err)
+	}
+	return nil
+}
+
+func (maps *DataMaps) DecodeLocalPerformanceData(filePath string) error {
+	file, err := os.Open(filePath)
+	if err != nil {
+		return fmt.Errorf("could not open local JSON file: %w", err)
+	}
+	defer file.Close()
+
+	var m map[string]json.RawMessage
```

Comment:
I'd prefer we properly define a struct and use `Unmarshal` here rather than using `RawMessage` + `Decode`

Reply:
Done, thanks for the suggestion. Its a lot better now.